peer/peerlist/list.go

```diff
@@ -373,7 +374,7 @@ func (pl *List) Choose(ctx context.Context, req *transport.Request) (peer.Peer,
 }
 
 func (pl *List) newNotRunningError(err error) error {
```

Comment:
if `err` is not a `*yarpcerrors.Status`, this will result in a different code than `CodeFailedPrecondition`, this should be changed

Reply:
ahh, correct! WaitUntilRunning was updated to return yarpcerrors explicitly.  We're not actually sure at this point if the errors is a Failed Precondition or a Invalid Request (WaitUntilRunning can return both).  
All the use cases where we are wrapping the error shooooould be using yarpcerrors, in the cases where the function does not return a yarpcerror it's probably desirable to mark that as "unknown" since we really don't know what happened.

This function is confusing though, I'll move the logic inline like the rest of the wrapper functions.